Python/compile.c

```diff
@@ -2065,6 +2079,72 @@ wrap_in_stopiteration_handler(struct compiler *c)
     return SUCCESS;
 }
 
+static int
+compiler_type_params(struct compiler *c, asdl_typeparam_seq *typeparams)
+{
+    if (!typeparams) {
+        return SUCCESS;
+    }
+    Py_ssize_t n = asdl_seq_LEN(typeparams);
+
+    for (Py_ssize_t i = 0; i < n; i++) {
+        typeparam_ty typeparam = asdl_seq_GET(typeparams, i);
+        location loc = LOC(typeparam);
+        switch(typeparam->kind) {
+        case TypeVar_kind:
+            ADDOP_LOAD_CONST(c, loc, typeparam->v.TypeVar.name);
+            if (typeparam->v.TypeVar.bound) {
+                expr_ty bound = typeparam->v.TypeVar.bound;
+                if (compiler_enter_scope(c, typeparam->v.TypeVar.name, COMPILER_SCOPE_TYPEPARAMS,
+                                        (void *)typeparam, bound->lineno) == -1) {
+                    return ERROR;
+                }
+                VISIT(c, expr, bound);
+                ADDOP_IN_SCOPE(c, loc, RETURN_VALUE);
+                PyCodeObject *co = optimize_and_assemble(c, 1);
+                compiler_exit_scope(c);
+                if (co == NULL) {
+                    return ERROR;
+                }
+                if (compiler_make_closure(c, loc, co, 0) < 0) {
+                    Py_DECREF(co);
+                    return ERROR;
+                }
+                Py_DECREF(co);
+                if (c->u->u_ste->ste_type_params_in_class) {
+                    ADDOP(c, loc, LOAD_LOCALS);
+                    ADDOP_I(c, loc, CALL_INTRINSIC_2, INTRINSIC_SET_CLASS_DICT);
+                }
```

Comment:
I understand from discussion yesterday that maybe there's a desire to move away from the oparg flags to MAKE_FUNCTION and instead just mutate the function after creation? I'm not sure about the pros/cons of that personally, but it still seems to me that that change, if we make it, should be orthogonal to this, and for this PR it's better to stay consistent. So I would prefer to see this set in `MAKE_FUNCTION` instead via a new oparg flag, for consistency with the current pattern, and because it avoids even needing the `SET_CLASS_DICT` intrinsic.

Reply:
I don't have a strong opinion here, but @markshannon asked me further up to switch to an intrinsic, so that's what I did.